Orb tools 11 (#6560)
circleci/circleci-docs: jekyll/_cci2/creating-orbs.md

## Introduction
{: #introduction }

After authoring your orb, you can publish it with a [semantically versioned]({{site.baseurl}}/2.0/orb-concepts/#semantic-versioning) tag, and the orb will appear on the [Orb Registry](https://circleci.com/developer/orbs).

> **Note:** If your orb is private, it will not be searchable on the Orb Registry, however the URL will be accessible to authenticated users of the orb.

Reviewer: ```suggestion
**Note:** If your orb is private, it will not be searchable on the Orb Registry. However, the URL will be accessible to authenticated users of the orb.
```

> ![Orb Publishing Process]({{ site.baseurl }}/assets/img/docs/orb-publishing-process.png)

Reviewer 2: @KyleTryon Do you think we need to reference private orbs everywhere? the line above says if you publish your orb will be in the registry 🤔

Author: You will be able to view your orb on the registry even if it is private if:
1. You are logged in and authenticated to the correct account
2. You know the exact URL or name of the orb.

You will not be able to search for the orb via the search box.

> ![Orb Publishing Process]({{ site.baseurl }}/assets/img/docs/orb-publishing-process.png)

Reviewer 2: @KyleTryon Do you think we need to reference private orbs everywhere? the line above says if you publish your orb will be in the registry 🤔

> ## Orb Development Kit

Author: Removed.

Next, follow these steps to create a new release from your changes.

1. **Open a new Pull Request to the default branch.** <br/>
Open a new Pull Request to the default branch. The included `config.yml` and `test-deploy.yml` files in the `./circleci` directory are configured to automatically [lint]({{site.baseurl}}/2.0/testing-orbs/#yaml-lint), [shellcheck]({{site.baseurl}}/2.0/testing-orbs/#shellcheck), [review]({{site.baseurl}}/2.0/testing-orbs/#review), and [test]({{site.baseurl}}/2.0/testing-orbs/#integration-testing) your orb changes on CircleCI.com

1. **Ensure all tests pass.** <br/>
You can view the results of your tests directly on GitHub within the Pull Request, or, for a more detailed view, watch the entire pipeline on CircleCI.com.

> Notice there are two workflows, `lint-pack` will run first and contains our linting, shellchecking, review, and will publish a development version to be tested in the second workflow. The `test-deploy` workflow will contain our integration tests, and can publish the production version of our orb when ready.

Reviewer 2: ```suggestion
Notice there are two workflows, `lint-pack` will run first and contains our linting, shellchecking, review, and will publish a development version to be tested in the second workflow. The `test-deploy` workflow contains our integration tests, and can publish the production version of our orb when ready.
```

 ![Orb test results as reported by GitHub Checks API on pull request]({{site.baseurl}}/assets/img/docs/orbtools-11-checks.png)

1. **"Squash" Merge.** <br/>
When your changes are complete, we recommend (not required) "[Squash Merging](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-pull-request-commits)" your changes into a single commit, with a [Conventional Commit Message](https://www.conventionalcommits.org/).

    Examples:

      - `fix: x-command parameter from string to integer`
      - `feat: added new x parameter to y command`

1. **Tag and Release!** <br/>
Your changes have now been merged to the default branch, but if you check the [Orb Registry](https://circleci.com/developer/orbs), you will see no new versions have been published.

> To publish a new version of our orb, we must tag our release. A tag can be created and pushed manually, however we recommend using [GitHub.com's Releases feature](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release).

Reviewer 2: ```suggestion
To publish a new version of our orb, you need to tag your release. A tag can be created and pushed manually, however we recommend using [GitHub.com's Releases feature](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release).
```

> Using GitHub's Releases feature will allow us to publish "Release Notes", which will function as a changelog for the new version of the orb.

Reviewer 2: ```suggestion
Using GitHub's Releases feature will allows you to publish "Release Notes", which will function as a changelog for your orb.
```

> Follow the GitHub docs to create a new release.

Reviewer 2: Need to link to the docs we want them to follow here

Reviewer 2: @KyleTryon

>    1. When asked to select a tag, specify the new tag that will be published with this release. Double check the current orb version in the [Orb Registry](https://circleci.com/developer/orbs). If you are unsure of what type of change you are making, we will have an opportunity to correct this in a moment. Ensure your tag fits the format `vX.Y.Z`

Reviewer 2: Are these supplementary steps to the GH docs we just asked them to follow, or a summary of those steps?

Reviewer 2: @KyleTryon
